pandas/tests/io/parser/test_python_parser_only.py

```diff
@@ -329,3 +332,73 @@ def readline(self):
             return self.data
 
     parser.read_csv(NoNextBuffer("a\n1"))
+
+
+@pytest.mark.parametrize("bad_line_func", [lambda x: ["2", "3"], lambda x: x[:2]])
+def test_on_bad_lines_callable(python_parser_only, bad_line_func):
+    # GH 5686
+    parser = python_parser_only
+    bad_sio = StringIO("a,b\n1,2\n2,3,4,5,6\n3,4")
```

Comment:
Don't have to change here, but I prefer:

data = """a,b
1,2
1,2,3
"""

Imho this is easier to read

Reply:
Sure changed.